pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -501,13 +503,33 @@ func (c *Cluster) DeployPrometheusRule(name, namespace string) error {
 	if err != nil {
 		return errors.Wrapf(err, "failed to set owner reference to prometheus rule %q", prometheusRule.Name)
 	}
+
 	cephv1.GetMonitoringLabels(c.spec.Labels).ApplyToObjectMeta(&prometheusRule.ObjectMeta)
+	applyDefaultMonitoringLabels(&prometheusRule.ObjectMeta)
+
 	if _, err := k8sutil.CreateOrUpdatePrometheusRule(prometheusRule); err != nil {
 		return errors.Wrap(err, "prometheus rule could not be deployed")
 	}
 	return nil
 }
 
+func applyDefaultMonitoringLabels(m *metav1.ObjectMeta) {
+	defaultLabels := map[string]string{
+		"prometheus": "rook-prometheus",
```

Comment:
previosly only the `PrometheusRule` objects had this label, now it's also applied to the `ServiceMonitor` objects. I think this makes more sense, as `ServiceMonitor`s are also selected via that label by default. 

Because this label wasn't set before on `ServiceMonitor`s, it could be set by the user - making it actually possible to use it with a non-rook-specific kube-prometheus stack. Since it can now be overridden, it's still safe.